starknet_py/contract_abi_resolver.py

```diff
@@ -0,0 +1,131 @@
+# Needed because of string typed Contract
+# pyright: reportUndefinedVariable=false
+import warnings
+from enum import Enum
+from typing import Union, List, TypedDict, Tuple
+
+from starknet_py.net.client import Client
+from starknet_py.net.client_errors import ContractNotFoundError, ClientError
+from starknet_py.net.client_models import Abi, DeclaredContract
+from starknet_py.net.models import Address
+from starknet_py.proxy_check import (
+    ProxyResolutionError,
+    OpenZeppelinProxyCheck,
+    ArgentProxyCheck,
+    ProxyCheck,
+)
+
+
+class ProxyConfig(TypedDict, total=False):
+    """
+    Proxy resolving configuration
+    """
+
+    proxy_checks: List[ProxyCheck]
+    """
+    List of classes implementing :class:`starknet_py.proxy_check.ProxyCheck` ABC,
+    that will be used for checking if contract at the address is a proxy contract.
+    """
+
+
+def prepare_proxy_config(proxy_config: Union[bool, ProxyConfig]) -> ProxyConfig:
```

Comment:
I'd prefer we have straightforward implementation here: proxy_config can be spread into simple arguments instead of union here, and we can have necessary logic for parsing `proxy_config` in Contract, like you suggested in the second point.